backend/src/routes/api/components/list.ts

```diff
@@ -1,34 +1,18 @@
 import { FastifyRequest } from 'fastify';
 import { KubeFastifyInstance, OdhApplication } from '../../../types';
-import { getRouteForApplication } from '../../../utils/componentUtils';
-import { getApplicationDefs, updateApplicationDefs } from '../../../utils/resourceUtils';
+import { getApplications, updateApplications } from '../../../utils/resourceUtils';
 
 export const listComponents = async (
   fastify: KubeFastifyInstance,
   request: FastifyRequest,
 ): Promise<OdhApplication[]> => {
-  const applicationDefs = getApplicationDefs();
-  const installedComponents = [];
+  const applications = await getApplications();
   const query = request.query as { [key: string]: string };
 
-  if (!query.installed) {
-    return await Promise.all(applicationDefs);
+  if (query.installed) {
+    return Promise.resolve(applications.filter((app) => app.spec.isEnabled));
```

Comment:
We need to determine if the application was disabled by the user at some previous point and not return those that were.

Reply:
done.